zerver/lib/users.py

```diff
@@ -511,6 +515,45 @@ def check_user_can_access_all_users(acting_user: Optional[UserProfile]) -> bool:
     return False
 
 
+def check_can_access_user(
+    target_user: UserProfile, user_profile: Optional[UserProfile] = None
+) -> bool:
+    if check_user_can_access_all_users(user_profile) or not user_access_restricted_in_realm(
+        target_user
+    ):
```

Comment:
We should do these checks in the opposite order, in that it'll save a `is_user_in_group` query to do do so.

Also probably would be better to not put them on the same line of code.

Reply:
Updated this.